Packs/DelineaSS/Integrations/DelineaSS/Delinea.py

```diff
@@ -0,0 +1,533 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+from typing import Dict
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
+    Should only do requests and return data.
+    """
+
+    def __init__(self, server_url: str, username: str, password: str, proxy: bool, verify: bool):
+        super().__init__(base_url=server_url, proxy=proxy, verify=verify)
+        self._username = username
+        self._password = password
+        self._token = self._generate_token()
+        self._headers = {'Authorization': self._token, 'Content-Type': 'application/json'}
+
+    def _generate_token(self) -> str:
+        """Generate an Access token using the user name and password
+        :return: valid token
+        """
+        body = {
+            "username": self._username,
+            "password": self._password,
+            "grant_type": "password"
+        }
+
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        return "Bearer " + (self._http_request("POST", "/oauth2/token", headers=headers, data=body)).get('access_token')
+
+    def getPasswordById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + str(secret_id) + "/fields/password"
+        return self._http_request("GET", url_suffix)
+
+    def getUsernameById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + str(secret_id) + "/fields/username"
+        return self._http_request("GET", url_suffix)
+
+    def getSecret(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + str(secret_id)
+
+        return self._http_request("GET", url_suffix)
+
+    def searchSecretIdByName(self, search_name: str) -> list:
+        url_suffix = "/api/v1/secrets/lookup?filter.searchText=" + search_name
+        response = self._http_request("GET", url_suffix).get('records')
+        idSecret = argToList(response)
+        search_id = []
+
+        if len(idSecret) != 0:
+            for element in idSecret:
+                getID = element.get('id')
+                search_id.append(getID)
+
+        return search_id
+
+    def searchSecret(self, **kwargs) -> list:
+        count_params = len(kwargs)
+        params = {}
+        if count_params > 0:
+            for key, value in kwargs.items():

```

Comment:
Please add a comment explaining the following replace changes.

Reply:
Resolved. We don't think there is need for comment just replacing _ with . for api.